hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/interfaces/BlockManager.java

```diff
@@ -34,31 +34,30 @@ public interface BlockManager {
    * Puts or overwrites a block.
    *
    * @param container - Container for which block need to be added.
-   * @param data     - Block Data.
+   * @param data - Block Data.
    * @return length of the Block.
-   * @throws IOException
    */
   long putBlock(Container container, BlockData data) throws IOException;
 
   /**
    * Puts or overwrites a block.
    *
    * @param container - Container for which block need to be added.
-   * @param data     - Block Data.
-   * @param incrKeyCount - Whether to increase container key count.
+   * @param data - Block Data.
+   * @param endOfBlock - The last putBlock call for this block (when
+   *                     all the chunks are written and stream is closed)
    * @return length of the Block.
-   * @throws IOException
    */
-  long putBlock(Container container, BlockData data, boolean incrKeyCount)
+  long putBlock(Container container, BlockData data, boolean endOfBlock)
       throws IOException;
 
   /**
    * Gets an existing block.
    *
-   * @param container - Container from which block need to be get.
+   * @param container - Container from which block needs to be fetched.
    * @param blockID - BlockID of the Block.
    * @return Block Data.
-   * @throws IOException
+   * @throws StorageContainerException when BcsId is unknown or mismatched
    */
   BlockData getBlock(Container container, BlockID blockID)
       throws IOException;
```

Comment:
this throws IOE only. if as of today it just throws ``StorageContainerException``, it doesn't mean tmrw it can't throw any IOE, I think the javadocs throws should be in sync with the method signature. In the description, we can add whatever we want to

Reply:
Thank you for the review! I have changed it, Please check if it is fine now